File: 8kyu/to-square-root-or-not-to-square-root/to-square-root-or-not-to-square-root.py
import math

def square_or_square_root(arr):
  def sq_rt(x):
    return math.sqrt(x)
  op = lambda num: sq_rt(num) if sq_rt(num).is_integer() else num**2
  return list(map(op, arr))

print(square_or_square_root([4, 3, 9, 7, 2, 1 ])) 
# [2, 9, 3, 49, 4, 1]

File: 8kyu/to-square-root-or-not-to-square-root/to-square-root-or-not-to-square-root.go
package main

import (
	"fmt"
	"math"
)

func SqRt(num int) float64 {
	return  math.Sqrt(float64(num))
}

func SquareOrSquareRoot(arr []int) (result []int){
	for _,num := range arr {
		sqrt := SqRt(num)
		
		if sqrt == math.Trunc(sqrt) {
			result = append(result, int(sqrt))
		} else {
			result = append(result, num * num)
		}
	}

	return result
}

func main() {
	fmt.Println(SquareOrSquareRoot([]int{4, 3, 9, 7, 2, 1}))
}
